tests/e2e-kubernetes/testsuites/credentials.go

```diff
@@ -243,50 +244,87 @@ func (t *s3CSICredentialsTestSuite) DefineTests(driver storageframework.TestDriv
+			overrideServiceAccountRole(ctx, f, sa, "")
 
-				sa := csiDriverServiceAccount(ctx, f)
-				overrideServiceAccountRole(ctx, f, sa, "")
+			framework.ExpectNoError(deleteCredentialSecret(ctx, f))
 
-				framework.ExpectNoError(deleteCredentialSecret(ctx, f))
+			// Trigger recreation of our pods to ensure they're not using deleted resources
+			killCSIDriverPods(ctx, f)
+		}
 
-				// Trigger recreation of our pods to ensure they're not using deleted resources
-				killCSIDriverPods(ctx, f)
+		var (
+			afterAllCleanup   []func(context.Context) error
+			oidcProvider      string
+			policyRoleMapping = map[string]*iamtypes.Role{}
+		)
+
+		BeforeAll(func(ctx context.Context) {
+			oidcProvider = oidcProviderForCluster(ctx, f)
+
+			By("Pre-creating IAM roles for common policies")
+			for _, policyARN := range []string{
+				iamPolicyS3FullAccess,
+				iamPolicyS3ReadOnlyAccess,
+				iamPolicyS3NoAccess,
+			} {
+				role, removeRole := createRole(ctx, f, assumeRolePolicyDocument(ctx), policyARN)
+				policyRoleMapping[policyARN] = role
+				afterAllCleanup = append(afterAllCleanup, removeRole)
 			}
+		})
 
-			policyRoleMapping := map[string]*iamtypes.Role{}
-			BeforeAll(func(ctx context.Context) {
-				By("Pre-creating IAM roles for common policies")
-				for _, policyARN := range []string{
-					iamPolicyS3FullAccess,
-					iamPolicyS3ReadOnlyAccess,
-					iamPolicyS3NoAccess,
-				} {
-					role, removeRole := createRole(ctx, f, assumeRolePolicyDocument(ctx), policyARN)
-					policyRoleMapping[policyARN] = role
-					afterAllCleanup = append(afterAllCleanup, removeRole)
-				}
-			})
+		AfterAll(func(ctx context.Context) {
+			By("Cleaning up resources created for credential tests")
+			cleanClusterWideResources(ctx)
 
-			AfterAll(func(ctx context.Context) {
-				By("Cleaning up resources created for Driver-level tests")
-				cleanClusterWideResources(ctx)
+			var errs []error
+			for _, f := range afterAllCleanup {
+				errs = append(errs, f(ctx))
+			}
+			framework.ExpectNoError(errors.NewAggregate(errs), "while cleanup global resource")
+		})
 
-				var errs []error
-				for _, f := range afterAllCleanup {
-					errs = append(errs, f(ctx))
-				}
-				framework.ExpectNoError(errors.NewAggregate(errs), "while cleanup global resource")
-			})
+		updateCSIDriversServiceAccountRole := func(ctx context.Context, policyARN string) {
+			By("Updating CSI Driver's Service Account Role")
+			sa := csiDriverServiceAccount(ctx, f)
+
+			role, removeRole := createRole(ctx, f, assumeRoleWithWebIdentityPolicyDocument(ctx, oidcProvider, sa), policyARN)
+			deferCleanup(removeRole)
 
+			sa, restoreServiceAccountRole := overrideServiceAccountRole(ctx, f, sa, *role.Arn)
+			deferCleanup(restoreServiceAccountRole)
+
+			waitUntilRoleIsAssumableWithWebIdentity(ctx, f, sa)
+
+			// Trigger recreation of our pods to use the new IAM role
+			killCSIDriverPods(ctx, f)
+		}
+
+		updateDriverLevelCredentials := func(ctx context.Context, policyARN string) {
```

Comment:
Can we call this something like `updateDriverLevelK8sSecret`? The current name doesn't make it obvious